cpp/include/cudf/datetime.hpp

```diff
@@ -133,6 +133,48 @@ std::unique_ptr<cudf::column> extract_second(
   cudf::column_view const& column,
   rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource());
 
+/**
+ * @brief  Extracts millisecond fraction from any date time type and returns an int16_t
+ * cudf::column.
+ *
+ * @param column cudf::column_view of the input datetime values
+ * @param mr Device memory resource used to allocate device memory of the returned column
+ *
+ * @returns cudf::column of the extracted int16_t milliseconds
+ * @throw cudf::logic_error if input column datatype is not TIMESTAMP
+ */
+std::unique_ptr<cudf::column> extract_millisecond_fraction(
+  cudf::column_view const& column,
+  rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource());
+
+/**
+ * @brief  Extracts microsecond fraction from any date time type and returns an int16_t
+ * cudf::column.
+ *
```

Comment:
I think an example here would be helpful. I'm not sure microsecond-fraction would be quickly understood.
```suggestion
 *
 * A microsecond fraction is only the 3 digits that make up the microsecond portion of a duration.
 * For example, the microsecond fraction of 1.234567890 seconds is 567.
 *
```